File: CSharp/Patterns/StripePattern.cs
using System;

namespace CSharp
{
    public class StripePattern : RTPattern
    {
        public StripePattern(Color a, Color b) : base(a, b)
        {
            // Nothing to do
        }

        public override Color PatternAt(Point p)
        {
            return (int) Math.Floor(p.x) % 2 == 0 ? a : b;
        }

        public override Color UVPatternAt(double u, double v, CubeFace face = CubeFace.None)
        {
            return Color.Black;
        }

        protected override Color PatternAtShape(RTObject obj, Point localPoint)
        {
            var patternPoint = TransformInverse * localPoint;
            return PatternAt(patternPoint);
        }
    }
}

File: Java/src/main/java/Patterns/StripePattern.java
package RTC.Patterns;

import RTC.Math.*;
import RTC.Material.*;

public class StripePattern extends Pattern {
    private Color a, b;

    public StripePattern(Color A, Color B) {
        super();

        a = A;
        b = B;
    }

    public Color patternAt(Point point) {
        if (Math.floor(point.getX()) % 2 == 0) {
            return a;
        } else {
            return b;
        }
    }
}